File: hello_ref.go
package main

import (
	"fmt"
	"os"
	"io"
)

var bufSize 	int 	= 1024
var filePath 	string 	= "./resources/readme.txt"

func main() {
	file, fileOpenError := os.Open(filePath)
	defer file.Close()

	err := fileOpenError
	buf := make([]byte, bufSize)
	for err != io.EOF {
		fmt.Printf("%s\n", string(buf))
		bytesRead, fileReadError := file.Read(buf)
		err = fileReadError
	}
}

File: HelloRef.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class HelloRef {
	private final static String FILE_PATH = "./resources/readme.txt";

	public static void main(String[] args) {
		BufferedReader bufReader;
		String line;
		try{
			bufReader = new BufferedReader(new FileReader(FILE_PATH));
			while ((line=bufReader.readLine())!=null){
				System.out.println(line);
			}
		}catch(IOException e){
			e.printStackTrace();
		}
	}
}
